File: jadx-gui/src/main/java/jadx/gui/plugins/script/ScriptCompletionRenderer.java
package jadx.gui.plugins.script;

import javax.swing.JList;

import org.fife.ui.autocomplete.Completion;
import org.fife.ui.autocomplete.CompletionCellRenderer;

import jadx.gui.settings.JadxSettings;

import static jadx.gui.utils.UiUtils.escapeHtml;
import static jadx.gui.utils.UiUtils.fadeHtml;
import static jadx.gui.utils.UiUtils.wrapHtml;

public class ScriptCompletionRenderer extends CompletionCellRenderer {

	public ScriptCompletionRenderer(JadxSettings settings) {
		setDisplayFont(settings.getFont());
	}

	@Override
	protected void prepareForOtherCompletion(JList list, Completion c, int index, boolean selected, boolean hasFocus) {
		ScriptCompletionData cmpl = (ScriptCompletionData) c;
		setText(wrapHtml(escapeHtml(cmpl.getInputText()) + "  "
				+ fadeHtml(escapeHtml(cmpl.getSummary()))));
	}
}

File: jadx-plugins/jadx-script-kotlin/src/main/kotlin/jadx/plugins/script/kotlin/gui/ScriptCompletionRenderer.kt
package jadx.plugins.script.kotlin.gui

import jadx.gui.settings.JadxSettings
import jadx.gui.utils.UiUtils
import org.fife.ui.autocomplete.Completion
import org.fife.ui.autocomplete.CompletionCellRenderer
import javax.swing.JList

class ScriptCompletionRenderer(settings: JadxSettings) : CompletionCellRenderer() {
	init {
		displayFont = settings.codeFont
	}

	override fun prepareForOtherCompletion(
		list: JList<*>?,
		c: Completion?,
		index: Int,
		selected: Boolean,
		hasFocus: Boolean,
	) {
		val cmpl = c as ScriptCompletionData
		setText(
			UiUtils.wrapHtml((UiUtils.escapeHtml(cmpl.inputText) + "  " + UiUtils.fadeHtml(UiUtils.escapeHtml(cmpl.summary)))),
		)
	}
}
